sealtk/core/test/ScalarFilterModel.cpp

```diff
@@ -0,0 +1,203 @@
+/* This file is part of SEAL-TK, and is distributed under the OSI-approved BSD
+ * 3-Clause License. See top-level LICENSE file or
+ * https://github.com/Kitware/seal-tk/blob/master/LICENSE for details. */
+
+#include <sealtk/core/ScalarFilterModel.hpp>
+
+#include <sealtk/core/AbstractItemModel.hpp>
+#include <sealtk/core/DataModelTypes.hpp>
+
+#include <vital/types/timestamp.h>
+
+#include <vital/range/iota.h>
+
+#include <QSet>
+
+#include <QtTest>
+
+namespace kvr = kwiver::vital::range;
+
+using time_us_t = kwiver::vital::timestamp::time_t;
+
+// ----------------------------------------------------------------------------
+template<> char* QTest::toString<QSet<int>>(QSet<int> const& set)
+{
+  QString s;
+
+  QDebug d{&s};
+  d << set;
+
+  return qstrdup(qPrintable(s));
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+namespace sealtk
+{
+
+namespace core
+{
+
+namespace test
+{
+
+namespace // anonymous
+{
+
+// ============================================================================
+class TestModel : public sealtk::core::AbstractItemModel
+{
+public:
+  TestModel(int rows) : rows{rows} {}
+
+  int rowCount(QModelIndex const& parent = {}) const override
+  { return (parent.isValid() ? 0 : this->rows); }
+
+  QVariant data(QModelIndex const& index, int role) const override;
+
+private:
+  int const rows;
+};
+
+// ----------------------------------------------------------------------------
+QVariant TestModel::data(QModelIndex const& index, int role) const
+{
+  if (this->checkIndex(index, IndexIsValid | ParentIsInvalid))
+  {
+    auto const row = index.row();
+    switch (role)
+    {
+      case sealtk::core::NameRole:
+      case sealtk::core::LogicalIdentityRole:
+        return row;
+
+      case sealtk::core::StartTimeRole:
+      case sealtk::core::EndTimeRole:
+        return QVariant::fromValue(static_cast<time_us_t>(row * 100));
+
+      default:
+        break;
+    }
+  }
+
+  return sealtk::core::AbstractItemModel::data(index, role);
```

Comment:
`this->`

Reply:
What is the actual rationale for this? I see *one* place that follows this pattern, currently, and several that don't.